engine/lib/membrane_rtc_engine/engine.ex

```diff
@@ -1140,8 +1141,19 @@ defmodule Membrane.RTC.Engine do
     end)
     |> Enum.flat_map(fn {endpoint_id, _endpoint} ->
       subscriptions = state.subscriptions[endpoint_id]
-      tracks = Enum.filter(tracks, &Map.has_key?(subscriptions, &1.id))
-      [notify_child: {{:endpoint, endpoint_id}, {:remove_tracks, tracks}}]
+
+      has_pending_sub? = fn track_id ->
+        state.pending_subscriptions
+        |> Stream.filter(&(&1.endpoint_id == endpoint_id))
+        |> Enum.any?(&(&1.track_id == track_id))
+      end
+
+      tracks =
+        Enum.filter(tracks, &(Map.has_key?(subscriptions, &1.id) or has_pending_sub?.(&1.id)))
```

Comment:
What do you think about creating a function name `is_subscribed?(state, endpoint_id, track_id)`?